### DISCUSSION TITLE
articles/2024/python-uv-revisited/

### DISCUSSION THREAD

**Discussion by giscus** [loopwerk/loopwerk.io#19] [Article discussions]
# articles/2024/python-uv-revisited/

Almost two months ago I compared Poetry with uv, and for me uv had some pretty significant drawbacks that kept me from switching over. The situation has changed quite a bit since then!

https://www.loopwerk.io/articles/2024/python-uv-revisited/

**Comment by montanarograziano** [comment]
Thank you for you article, I've been reading all the comparison among uv vs poetry vs pdm, and I too am in the need of migrating lots of projects from poetry to uv. In your experience, is there any automated procedure that could help in task? Or does every project need to be handled specifically? Did you find any helpful insight in your migration? I think it would be helpful to write something down as it may serve other users aswell.

**Reply by kevinrenskers** [reply]
Sadly there is no fully automatic migration just yet, but it's not too bad: `uvx pdm import pyproject.toml` will get you 95% of the way there, and then you just need to make minor adjustments by hand, mainly removing old Poetry stuff from the file. It took me maybe 2 minutes per project.

See https://github.com/astral-sh/uv/issues/5200#issuecomment-2468319391 for detailed instructions.

**Reply by montanarograziano** [reply]
Thanks, I was trying to do exactly that procedure, I'll se how it goes.

**Comment by ratulotron** [comment]
Thanks for the comparisons, finally switched to `uv` in a small production project. If `uv` could bump the pinned versions of dependencies in the pyproject.toml that would have been neat!

**Comment by Winand** [comment]
```bash
uv add --group prod --no-sync gunicorn
```
I don't get it why when add a package to a group and it is installed into the evironment (when `--no-sync` is NOT specified).
Then I run `uv sync` and it is removed from the environment.

**Reply by Winand** [reply]
Sorry, I've wanted to fix grammar but completely failed with this task:-)

**Comment by nogjam** [comment]
At the end of your [previous article](https://www.loopwerk.io/articles/2024/python-poetry-vs-uv/) you linked to [others](https://mastodon.social/@glyph/113093806840686501) who talked about the sustainability of the project as a concern. I don't really see that concern addressed in this article, though it seems your primary concerns were otherwise. Do you have any further thoughts on the sustainability aspect?

**Reply by kevinrenskers** [reply]
It's still a concern, but indeed it wasn't my primary one. I decided that I can live with this concern considering the upsides of switching to uv. I did address that in this article, right at the top:

> Yes, the smaller drawbacks are unchanged: uv is not written in Python, locking out most of the Python community from contributing; Astral, the company behind uv doesn’t seem to have a way to monetize their work; and I think the command interface for uv still doesn’t make much sense. But I can forget about all of that when uv has such clear benefits – especially its ability to manage the installed Python version.
